### PR TITLE
Make your last hitter get a kill if you suicide any possible way

### LINKED ISSUE #756: It is possible to hit and kill yourself
I discovered it a while ago, and it leads to very strange kill messages. I mean how is that even possible? Should you get the message "You are in your team!" ?

How to reproduce:
Using a dragonfire script:
```lua
minetest.register_globalstep(function()
	local player = minetest.localplayer
	if not player then return end
	local playercao = player:get_object()
	local control = player:get_control()
	if control.dig then
		playercao:punch()
	end
end)
```
Do not ask me how I found this out.

### PR DESCRIPTION
**Pull request by savilli**
Fix #756

### CODE REVIEW

File: mods/pvp/anticoward/init.lua [hunk 1]
```diff
@@ -15,18 +15,7 @@ ctf.register_on_attack(function(player, hitter,
 		local pname = player:get_player_name()
 		local hname = hitter:get_player_name()
 
-		local hp = player:get_hp() - damage
-		if hp <= 0 then
```
Comment (savilli) [inline mods/pvp/anticoward/init.lua:19]: Dead code since on_attack is not called if player:get_hp() - damage <= 0

**Comment by savilli** [comment]
Testing would be nice
Test cases:
- Honest death in battle 
- Combatlogging
- Run from a battle and suicide using fail damage
- Run from a battle and suicide using a grenade

**Comment by xenonca** [comment]
Why would the last hitter get the points? I don't know if that is very fair in every case. I'd prefer a solution where you're simply unable to kill yourself (unless it's a nade or so)

**Comment by savilli** [comment]
> Why would the last hitter get the points? I don't know if that is very fair in every case. I'd prefer a solution where you're simply unable to kill yourself (unless it's a nade or so)

How it's not fair? If you die by falling your your last hitter will get kill. Kill yourself using a nade is a suicide too. What is the reason why it is not the same?

**Comment by savilli** [comment]
Just a quick summary from the discussion.

In the old days if you didn't want to give your enemy a kill you could run away from a battle and suicide. Anticoward system was designed to prevent such tactic. If you suicide using fail damage your last hitter will get a kill.
But if you die from hitting yourself or nading yourself your enemy still doesn't get a kill because anticoward system doesn't see it as a suicide. This pr will make that your last hitter get a kill if you suicide ANY possible way.

You can forbid players to hit themself but after this pr it will be completely useless so I don't see a reason to do it.

**Comment by xenonca** [comment]
> You can forbid players to hit themself but after this pr it will be completely useless so I don't see a reason to do it.

Well, because I think that that's the proper solution, to prevent players from hitting themselves. Also it might not necessarily be useless if we find another way of suiciding we haven't thought of yet

**Comment by savilli** [comment]
> prevent players from hitting themselves.

No need to prevent players from hitting themselves just because. I don't think that hitting yourself and nading yourself are different and need different approach to handle.

**Comment by realOneplustwo** [comment]
Grenading yourself was, I believe, an intended consequence of adding grenades. Being able to hit yourself is not an intended feature, the two are vastly different.

**Comment by realOneplustwo** [comment]
What problem does this PR solve that cannot be solved through simpler means?

**Comment by savilli** [comment]
> Grenading yourself was, I believe, an intended consequence of adding grenades. Being able to hit yourself is not an intended feature, the two are vastly different.

It's not relevant
> What problem does this PR solve that cannot be solved through simpler means?

This PR is pretty simple

**Comment by xenonca** [comment]
It is very relevant as killing yourself with a nade (ideally killing some enemies with it too) can be a tactical reason to suicide. On the other side a PR is usually not good **just** because it's simple.

**Comment by realOneplustwo** [comment]
The simplest way to fix #756 is simply to not let the player hit herself.

**Comment by savilli** [comment]
> The simplest way to fix #756 is simply to not let the player hit herself.

It's more complicated since you need to distinguish hitting yourself using a sword or a grenade

**Pull request closed without merging**